price_feed_poller.py

```diff
@@ -20,75 +18,101 @@ def handle_requestUpdate():
 
     print(f"Got {balance} wei")
 
-
-    dr_id = pricefeed.functions.requestUpdate().transact(
+    # Hardcoded gas since it does not estimate well
+    dr_id = pricefeedcontract.functions.requestUpdate().transact(
         {"from": account_addr, "gas": 310000, "value": 310000000000000})
-    post_tx_hash = bytes(dr_id)
-    w3.eth.waitForTransactionReceipt(dr_id)
-    requestId = pricefeed.functions.lastRequestId().call()
 
-    print(
+     # Get receipt of the transaction   
+    receipt = w3.eth.waitForTransactionReceipt(dr_id)
+
+    # Check if transaction was succesful
+    if receipt['status']: 
```

Comment:
In case of failure does `receipt['status']` return `False` or does it raise a `KeyError`?

And add log for error case, or raise an exception and handle by caller.

Reply:
A false statement. I will add a log